packages/flame/lib/src/geometry/collision_detection.dart

```diff
@@ -52,8 +95,16 @@ Set<Vector2> intersections(
         shapeA.onCollision(currentResult, shapeB);
         shapeB.onCollision(currentResult, shapeA);
         currentResult.clear();
+        _shapeHashes.add(_combineHashCodes(shapeA, shapeB));
+      } else {
+        _handleShapeCollisionEnd(shapeA, shapeB);
       }
     }
   }
   return result;
 }
+
+// Note that this might result in hash collisions
```

Comment:
This is an interesting approach, butI imagine that this collision could result in some problems right? Or is this not something to worry about?

Reply:
Came up with a better idea.